rename settings.local.json to settings.json

zieka opened the pull request:
`.claude/settings.local.json` is intended for local use and not ment to be committed while `.claude/settings.json` is meant to be checked in

Use this link to re-run the recipe: https://app.moderne.io/recipes/org.openrewrite.RenameFile?organizationId=QUxML09wZW4gU291cmNlL09wZW5SZXdyaXRl#defaults%3DW3sidmFsdWUiOiIuY2xhdWRlL3NldHRpbmdzLmxvY2FsLmpzb24iLCJuYW1lIjoiZmlsZU1hdGNoZXIifSx7InZhbHVlIjoic2V0dGluZ3MuanNvbiIsIm5hbWUiOiJmaWxlTmFtZSJ9XQ%3D%3D

timtebeek commented:
I'd prefer to keep `RemoveDoublyAnnotatedCodehausAnnotations` as a recipe that merely removes annotations, and does not itself do any conversion.

Can we move this visitor into a either the existing `JsonIncludeAnnotation` recipe, or a standalone recipe?

timtebeek commented on src/main/java/org/openrewrite/java/jackson/codehaus/TransferUsingArgumentFromCodehausToFasterXML.java:
```diff
@@ -0,0 +1,117 @@
+
+import java.util.*;
+
+public class TransferUsingArgumentFromCodehausToFasterXML extends Recipe {
```
Be sure to also link it into the relevant yaml recipe in src/main/resources

timtebeek reviewed (commented):
Thanks for the work & rework here @nielsdebruin ; ended up removing the troublesome auto format; when in doubt that's typically easiest.

timtebeek commented on src/test/java/org/openrewrite/java/jackson/codehaus/TransferUsingArgumentFromCodehausToFasterXMLTest.java:
```diff
@@ -0,0 +1,93 @@
+              @JsonSerialize(include = NON_NULL, using = None.class)
+              @com.fasterxml.jackson.databind.annotation.JsonSerialize(using = None.class)
+              class Test {
+                @JsonSerialize(include = NON_NULL, using = None.class)
+                @com.fasterxml.jackson.databind.annotation.JsonSerialize(using = None.class)
```
What I found slightly odd about the "after" usage example here is that we're one-to-one moving the argument over, whereas the `using` argument here, as well as the `contentUsing`, `keyUsing` and `nullsUsing` sibling only accept a FasterXml JsonSerializer, whereas the moved argument is a Codehaus JsonSerialzer, That means after this recipe the code would not yet compile. Compare the before and after packages:

![image](https://github.com/user-attachments/assets/3127a4ce-0b17-46e0-b65c-b69ec9613e67) ![image](https://github.com/user-attachments/assets/fc44fbcf-016a-4f6b-8dae-7f03963d57d7)

That means we likely need to expand the converted packages (ideally) and types (for exceptional cases) seen here:
https://github.com/openrewrite/rewrite-jackson/blob/b5fab130a8a7a90774582b1eda0ec0b58217bab6/src/main/resources/META-INF/rewrite/codehaus-to-fasterxml.yml#L32-L62

timtebeek replied:
Pushed up in https://github.com/openrewrite/rewrite-jackson/pull/16/commits/736fce00e97139dd7a30f5aa529d6bb0c3e575d2.

timtebeek reviewed (approved):
Added a few more polishes, like also converting known serializer packages to fasterxml databind. Thanks for getting this most of the way there!

timtebeek reviewed (changes requested):
Before we merge; would you mind handling the other using arguments the same?
> contentUsing, keyUsing and nullsUsing

nielsdebruin commented:
> Before we merge; would you mind handling the other using arguments the same?
> 
> > contentUsing, keyUsing and nullsUsing

On it

nielsdebruin commented on src/main/java/org/openrewrite/java/jackson/codehaus/TransferJsonSerializeArgumentsFromCodehausToFasterXML.java:
```diff
@@ -0,0 +1,127 @@
+                        // Map from codehaus -> fasterxml annotation
+                        Map<J.Annotation, J.Annotation> doubleAnnotated = new FindDoublyAnnotatedVisitor().reduce(tree, new HashMap<>());
+
+                        transferArgument(doubleAnnotated, "using");
```
Done

zieka committed:
rename settings.local.json to settings.json

Use this link to re-run the recipe: https://app.moderne.io/recipes/org.openrewrite.RenameFile?organizationId=QUxML09wZW4gU291cmNlL09wZW5SZXdyaXRl#defaults%3DW3sidmFsdWUiOiIuY2xhdWRlL3NldHRpbmdzLmxvY2FsLmpzb24iLCJuYW1lIjoiZmlsZU1hdGNoZXIifSx7InZhbHVlIjoic2V0dGluZ3MuanNvbiIsIm5hbWUiOiJmaWxlTmFtZSJ9XQ%3D%3D

Co-authored-by: Moderne <team@moderne.io>
